addons/project/tests/test_project_recurrence.py

```diff
@@ -458,3 +458,62 @@ def test_recurrence_next_dates_year(self):
         self.assertEqual(dates[2], datetime(2023, 11, 30))
         self.assertEqual(dates[3], datetime(2024, 11, 30))
         self.assertEqual(dates[4], datetime(2025, 11, 30))
+
+    def test_recurrence_cron_repeat_after_subtasks(self):
+        parent_task = self.env['project.task'].create({
+            'name': 'Parent Task',
+            'project_id': self.project_recurring.id
+        })
+        domain = [('project_id', '=', self.project_recurring.id)]
+        domain_without_parent = [('project_id', '=', self.project_recurring.id), ('parent_id', '!=', False)]
+        with Form(parent_task.with_context({'tracking_disable': True})) as task_form:
+            with task_form.child_ids.new() as subtask_form:
+                subtask_form.name = 'Test Subtask 1'
+        with freeze_time("2020-01-01"):
+            with Form(parent_task.child_ids.with_context({'tracking_disable': True})) as form:
+                form.description = 'my super recurring task bla bla bla'
+                form.date_deadline = datetime(2020, 2, 1)
+
+                form.recurring_task = True
+                form.repeat_interval = 1
+                form.repeat_unit = 'month'
+                form.repeat_type = 'after'
+                form.repeat_number = 2
+                form.repeat_on_month = 'date'
+                form.repeat_day = '15'
+            subtask = form.save()
+            subtask.planned_hours = 2
+
+            self.assertEqual(subtask.recurrence_id.next_recurrence_date, date(2020, 1, 15))
+            self.assertEqual(self.env['project.task'].search_count(domain), 2)
+            self.assertEqual(self.env['project.task'].search_count(domain_without_parent), 1)
+            self.env['project.task.recurrence']._cron_create_recurring_tasks()
+            self.assertEqual(self.env['project.task'].search_count(domain), 2, 'no extra task should be created')
+            self.assertEqual(subtask.recurrence_id.recurrence_left, 2)
+
+        with freeze_time("2020-01-15"):
+            self.assertEqual(self.env['project.task'].search_count(domain), 2)
+            self.assertEqual(self.env['project.task'].search_count(domain_without_parent), 1)
+            self.env['project.task.recurrence']._cron_create_recurring_tasks()
+            self.assertEqual(self.env['project.task'].search_count(domain), 3)
+            self.assertEqual(self.env['project.task'].search_count(domain_without_parent), 2)
+            self.assertEqual(subtask.recurrence_id.recurrence_left, 1)
+
+        with freeze_time("2020-02-15"):
+            self.env['project.task.recurrence']._cron_create_recurring_tasks()
+            self.assertEqual(self.env['project.task'].search_count(domain), 4)
+            self.assertEqual(self.env['project.task'].search_count(domain_without_parent), 3)
```

Comment:
Use the method in the previous suggestion to reduce the number of query by one. 🙂